src/PollingBlockTracker.test.ts

```diff
@@ -0,0 +1,3024 @@
+          recordCallsToSetTimeout({ numAutomaticCalls: 1 });
+
+          await withPollingBlockTracker(
+            {
+              provider: {
+                stubs: [
+                  {
+                    methodName: 'eth_blockNumber',
+                    implementation: () => {
+                      throw 'boom';
+                    },
+                  },
+                  {
+                    methodName: 'eth_blockNumber',
+                    response: {
+                      result: '0x0',
+                    },
+                  },
+                ],
+              },
+            },
+            async ({ blockTracker }) => {
+              jest.spyOn(console, 'error').mockImplementation(EMPTY_FUNCTION);
+
+              blockTracker[methodToAddListener]('latest', EMPTY_FUNCTION);
+              await new Promise((resolve) => {
+                blockTracker.on('_waitingForNextIteration', resolve);
+              });
+
+              expect(console.error).toHaveBeenCalledWith(
+                expect.objectContaining({
+                  message: expect.stringMatching(
+                    /^PollingBlockTracker - encountered an error while attempting to update latest block:\nboom/u,
+                  ),
+                }),
+              );
+            },
+          );
+        });
+
+        it('should log an error if, while making the request for the latest block number, the provider rejects with an error and there is nothing listening to "error"', async () => {
+          recordCallsToSetTimeout({ numAutomaticCalls: 1 });
+
+          await withPollingBlockTracker(
+            {
+              provider: {
+                stubs: [
+                  {
+                    methodName: 'eth_blockNumber',
+                    error: 'boom',
+                  },
+                  {
+                    methodName: 'eth_blockNumber',
+                    response: {
+                      result: '0x0',
+                    },
+                  },
+                ],
+              },
+            },
+            async ({ blockTracker }) => {
+              jest.spyOn(console, 'error').mockImplementation(EMPTY_FUNCTION);
+
+              blockTracker[methodToAddListener]('latest', EMPTY_FUNCTION);
+              await new Promise((resolve) => {
+                blockTracker.on('_waitingForNextIteration', resolve);
+              });
+
+              expect(console.error).toHaveBeenCalledWith(
+                expect.objectContaining({
+                  message: expect.stringMatching(
+                    /^PollingBlockTracker - encountered an error while attempting to update latest block:\nError: boom/u,
+                  ),
+                }),
+              );
+            },
+          );
+        });
+
+        it('should update the current block number', async () => {
```

Comment:
This seems like a dupe of the `should take a listener that is called soon after being added` test on line 738

Reply:
Oh I see, this test checks the current block v.s. the return value. Nevermind!

Reply:
It's close, but this is testing the result of `getCurrentBlock` and not what argument the "latest" event is emitted with.